src/dbos.ts

```diff
@@ -1241,6 +1243,21 @@ export class DBOS {
 
       const wfctx = assertCurrentWorkflowContext();
 
+      // Set child deadline if a timeout is configured
+      // Explicitly set to undefined means "detach the child from the parent's deadline"
+      // We must check the inParams but also the pctx (if the workflow was called withWorkflowTimeout)
+      let deadlineEpochMS: number | undefined = wfctx.deadlineEpochMS;
+      if (Object.prototype.hasOwnProperty.call(inParams || {}, 'timeoutMS') && inParams?.timeoutMS === undefined) {
```

Comment:
As Peter said above, we should use `null` to indicate `clear the timeout`, which should simplify this code signficantly.